From ae99ac32f61e0db92a45179579030a23fe1b5770 Mon Sep 17 00:00:00 2001
From: Peter Harris <pharris2@rocketsoftware.com>
Date: Tue, 15 Oct 2024 09:51:33 -0400
Subject: [PATCH] Fix build when HAVE_STRCASECMP is not defined

Regression introduced in commit 4792e9e798de327572aba1575438b6936a55c7ef
"Clear numerous -Wsign-compare warnings from gcc 14.1".

Part-of: <https://gitlab.freedesktop.org/xorg/lib/libxtrans/-/merge_requests/22>
---
 Xtrans.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Xtrans.c b/Xtrans.c
index c595bc2..82ab218 100644
--- a/Xtrans.c
+++ b/Xtrans.c
@@ -162,7 +162,7 @@ TRANS(SelectTransport) (const char *protocol)
     strncpy (protobuf, protocol, PROTOBUFSIZE - 1);
     protobuf[PROTOBUFSIZE-1] = '\0';
 
-    for (i = 0; i < PROTOBUFSIZE && protobuf[i] != '\0'; i++)
+    for (unsigned int i = 0; i < PROTOBUFSIZE && protobuf[i] != '\0'; i++)
 	if (isupper ((unsigned char)protobuf[i]))
 	    protobuf[i] = tolower ((unsigned char)protobuf[i]);
 #endif
-- 
GitLab

